### DISCUSSION TITLE
Could not load database driver elasticsearch+http

### DISCUSSION THREAD

**Discussion by davidsheard** [apache/superset#19189] [Q&A / Help]
Hi,
I have spent all day trying to connect Superset to Elasticsearch
I added a Database and tested the connection using this string:
elasticsearch+http://elastic:elastic@localhost:9200/
  I can connect to Kibana using elastic:elastic as I set this password. I also removed basic security on elasticsearch and ran:
 elasticsearch+http://localhost:9200/ - with same error.

I get this error: 
"ERROR: Could not load database driver: elasticsearch+http"
I even removed the training forward slash.

I am running a local instance of Superset and Elasticsearch on my Mac.
I am using Docker and Superset is running fine:
docker-compose -f docker-compose.yml up. (I even tried docker-compose  up)
superset_node            | No type errors found
superset_node            | Version: typescript 4.0.3
superset_node            | Time: 119888ms
superset_node            | <s> [webpack.Progress] 100% 
superset_node            | 
superset_node            |    10612 modules

I installed:
pip install elasticsearch-dbapi
I tried elasticsearch 7.6.2 and then installed 7.10.1 which is running fine.
pip install elasticsearch==7.10.1

I tired with basic security via X-Pack and without.
I have no idea what to try next?? If I can't find a solution I'll go back to Kiabna. I am playing to see if it's worth adding to our visualisation tools at our company but so far not looking good.

**Comment by davidsheard** [comment]
Extra information:
ip freeze | grep elasticsearch
elasticsearch==7.10.1
elasticsearch-dbapi==0.2.0

**Comment by BFergerson** [comment]
I've only just started looking into how to use superset and the first issue I ran into has to do with the missing `docker-compose-non-dev.yml` file as mentioned in https://superset.apache.org/docs/installation/installing-superset-using-docker-compose. I've got superset running by instead just running `docker-compose up` in the `docker` directory.

This is the issue I'm at now. Superset looks really nice but the documentation seems to be a bit erratic for those getting started.
I'll update this if I figure it out.

**Comment by BFergerson** [comment]
Looks like I was just dealing with old documentation. I was able to setup elasticsearch after following https://superset.apache.org/docs/databases/dockeradddrivers.

**Comment by nytai** [comment]
@BFergerson If you checkout `master` the `docker-compose-non-dev.yml` should exist. The documentation was published before an official release, hence the file is missing from the `latest` tag.

**Comment by sarcoex** [comment]
Hello, I still have this issue, the "docker-compose-non-dev.yml" exists.

The error on the superset page is: "An error occurred while creating databases: (builtins.NoneType) None
[SQL: Error connecting to Elasticsearch]
(Background on this error at: https://sqlalche.me/e/14/dbapi)"
